File: Python/InsertionSort.py
def swap(array, idx0, idx1):
    temp = array[idx0]
    array[idx0] = array[idx1]
    array[idx1] = temp
    return array


arr = [9, 8, 7, 6, 5, 5, 4, 3, 2, 1, 0]

for i in range(1, len(arr)):
    j = i
    while j > 0 and arr[j-1] > arr[j]:
        arr = swap(arr, j, j-1)
        j -= 1

print(arr)

File: Java/InsertionSort.java
import java.util.Arrays;

public class InsertionSort
{
    public static void main(String [] args) {
        int[] arr = new int[]{9, 8, 7, 6, 5, 5, 4, 3, 2, 1, 0};
        
        for(int i=1;i<arr.length;i++) {
            int j = i;
            while(j > 0 && arr[j-1] > arr[j]) {
                arr = swap(arr, j, j-1);
                j -= 1;
            }
        }
        
        System.out.println(Arrays.toString(arr));
    }
    
    public static int[] swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
        return arr;
    }
}